tools/fileconv/api3/parameters.go

```diff
@@ -45,8 +46,24 @@ func CamelCaseToSpaceSeparated(displayName string) string {
 	return strcase.ToDelimited(displayName, ' ')
 }
 
+// ParameterFilterGetMethod callback that filters REST operations based on endpoint parameters.
+type ParameterFilterGetMethod func(objectName string, operation *openapi3.Operation) bool
+
+// OnlyOptionalGetParameters operation must include only optional parameters.
+func OnlyOptionalGetParameters(objectName string, operation *openapi3.Operation) bool {
+	for _, parameter := range operation.Parameters {
+		if parameter.Value.Required {
```

Comment:
<img width="980" alt="Screenshot 2024-09-02 at 12 58 18 PM" src="https://github.com/user-attachments/assets/741d5b00-151a-4898-9377-a3080fa84900">
OpenAPI calls these as parameters, and it seems like this code would skip any required parameters in the types listed above. 

I think to avoid unexpected behavior, we can introduce a simpler filter method that filters out parameters based on their name

Reply:
I do remember thinking/addressing it.
I checked other code on different branches and I don't seem to find it, which means I was thinking and forgot :cry: 
Will rename the method to be more explicit

Reply:
I kept the signutre very general and I am passing operation not even parameters to give more context. Object name comes alongside to address special cases if ever.

Reply:
Usage example by gong:
```
api3.WithParameterFilterGetMethod(func(objectName string, operation *openapi3.Operation) bool {
  if objectName == "calls" {
    // Even though OpenAPI docs and official documentation say that some query parameters are required
    // in practice you still can make an API call without any specified.
    // Must include this object.
    return true
  }

  return api3.OnlyOptionalQueryParameters(objectName, operation)
}),
```